gitops/snapshot.go

```diff
@@ -169,7 +169,12 @@ func SetSnapshotIntegrationStatusAsInvalid(snapshot *applicationapiv1alpha1.Snap
 		Message: message,
 	}
 	meta.SetStatusCondition(&snapshot.Status.Conditions, condition)
-	go metrics.RegisterInvalidSnapshot(condition.Type, condition.Reason)
+}
+
+// SetSnapshotIntegrationStatusAsInvalidandNoRequeue sets the AppStudio integration status condition for the Snapshot to invalid and it'll not be requeued.
+func SetSnapshotIntegrationStatusAsInvalidandNoRequeue(snapshot *applicationapiv1alpha1.Snapshot, message string) {
+	SetSnapshotIntegrationStatusAsInvalid(snapshot, message)
+	go metrics.RegisterInvalidSnapshot(AppStudioIntegrationStatusCondition, AppStudioIntegrationStatusInvalid)
```

Comment:
What's the difference between this line here and [this one](https://github.com/redhat-appstudio/integration-service/blob/main/gitops/snapshot.go#L172) in `SetSnapshotIntegrationStatusAsInvalid()`?

Reply:
No difference.  I removed that line from SetSnapshotIntegrationStatusAsInvalid()  since  the Snapshot will be requeued in the situation. Then Snapshot is not in the final status and will not be recorded.

Reply:
Got it, thanks for the clarification!